x-pack/auditbeat/processors/sessionmd/add_session_metadata.go

```diff
@@ -96,13 +96,24 @@ func New(cfg *cfg.C) (beat.Processor, error) {
 }
 
 func (p *addSessionMetadata) Run(ev *beat.Event) (*beat.Event, error) {
-	_, err := ev.GetValue(p.config.PIDField)
+	pi, err := ev.GetValue(p.config.PIDField)
 	if err != nil {
 		// Do not attempt to enrich events without PID; it's not a supported event
 		return ev, nil //nolint:nilerr // Running on events without PID is expected
 	}
 
-	err = p.provider.UpdateDB(ev)
+	// Do not enrich failed syscalls, as there was no actual process change related to it
+	v, err := ev.GetValue("auditd.result")
+	if err == nil && v == "fail" {
```

Comment:
so we don't care here if `err != nil`?

Reply:
no, we don't. If this processor were to run on a non-auditd process event, it's fine if the auditd fields don't exist. Ideally this will work on any event that has a PID field (although that probably not exactly true right now)